app/controllers/api/v1/user_roles_controller.rb

```diff
@@ -12,11 +20,36 @@ def show
         render json: @user_role, serializer: serializer
       end
 
+      def destroy
+        current_user.remove_role_for_user(@user_role)
+        render json: {}
+      end
+
+      api :POST, "/v1/user_roles", "Add an user_role"
+      param_group :user_role
+      def create
+        @user_role = current_user.assign_role_for_user(
+          user_id: params["user_role"]["user_id"],
+          role_id: params["user_role"]["role_id"],
+          expires_at: params["user_role"]["expires_at"]
+        )
```

Comment:
Same as above, don't think creating an instance member is needed.

Note: you can use the `slice` method on the hash to pick multiple properties at once, e.g

```suggestion
        user_role = current_user.assign_role_for_user(
           params["user_role"].slice('user_id', 'role_id', 'expires_at')
        )
```

not a big deal, just worth mentioning : )